web/src/features/Dashboard/api/getUpdateDownloadStatus.ts

```diff
@@ -28,32 +22,27 @@ const getUpdateDownloadStatus = async (
   );
 
   if (!res.ok) {
-    throw new Error("Error getting update status");
+    throw new Error("failed to get rewrite status");
   }
   const appResponse = await res.json();
-  return appResponse;
-};
 
-const makeUpdateStatusResponse = (
-  response: UpdateStatusResponse
-): UpdateStatus => {
-  return {
-    checkingForUpdateError: response.status === "failed",
-    checkingForUpdates: response.status !== "running",
-    checkingUpdateMessage: response.currentMessage,
-    status: response.status,
-  };
+  return appResponse;
 };
 
-export const useUpdateDownloadStatus = () => {
+export const useUpdateDownloadStatus = (
+  onSuccess: (data: UpdateStatusResponse) => void,
+  onError: (error: Error) => void,
+  isBundleUploading: boolean
+) => {
   const { selectedApp } = useSelectedApp();
 
   return useQuery({
     queryFn: () => getUpdateDownloadStatus(selectedApp?.slug || ""),
     queryKey: ["getUpdateStatus"],
-    onError: (err: Error) => console.log(err),
-    refetchInterval: (data) => (data?.status !== "running" ? false : 1000),
-    select: makeUpdateStatusResponse,
+    onSuccess,
```

Comment:
onSuccess is passed in and will set a few states. https://github.com/replicatedhq/kots/blob/3e592931e7c009c2bf8b00de561b2bae9afe92a3/web/src/features/Dashboard/components/Dashboard.tsx#L251
onError is the same.